backend/services/implementations/classService.ts

```diff
@@ -198,6 +199,10 @@ class ClassService implements IClassService {
           `Class with id ${id} not found or not currently active`,
         );
       }
+
+      // Make a best effort to delete upcoming test sessions and end active test sessions
+      await this.deleteUpcomingTestSessions(id, date);
+      await this.endActiveTestSessions(id, date);
```

Comment:
why do we need a `date` parameter at all? can we not just get the current date in the functions themselves? or is there logic here i'm missing

Reply:
i wanted to use the same date for both functions and in the various places its used for consistency